### PR TITLE
MULE-9057: Flag when Deployment Service has finished deploying all ex…

### PR DESCRIPTION
**Pull request by eguser**
…isting domains and applications upon Mule startup

### CODE REVIEW

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java [hunk 1]
```diff
@@ -104,6 +106,7 @@ public void start()
             try
             {
                 listener.onAfterStartup();
+                this.isStarted = true;
```
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: I think you should set isStarted to false when stopped.

**Comment by asequeira** [comment]
Please add a test case

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java (hunk 1 above)
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: Shouldn't this be set before the listeners? I mean, I would expect the isStarted to be true if I'm executing onAfterStartup() method.

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java [hunk 2]
```diff
@@ -118,6 +121,10 @@ public void stop()
         deploymentDirectoryWatcher.stop();
     }
 
+    public boolean isStarted() {
```
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: Fix identation.

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java [hunk 3]
```diff
@@ -118,6 +121,10 @@ public void stop()
...
+        return this.isStarted;
```
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: Remove "this"

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java (hunk 1 above)
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: Remove "this"

File: modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java [hunk 4]
```diff
@@ -65,6 +65,8 @@
     private final DeploymentDirectoryWatcher deploymentDirectoryWatcher;
     private DefaultArchiveDeployer<Application> applicationDeployer;
 
+    private static boolean isStarted = false;
```
Comment (asequeira) [inline modules/launcher/src/main/java/org/mule/module/launcher/MuleDeploymentService.java]: Shouldn't this be an instance variable?

**Comment by eguser** [comment]
Thanks Ale,

I have implemented your suggestions and have added a test case.

**Comment by eguser** [comment]
Would isReady() and "boolean deploymentServiceReady" make more sense? Is there an existing way to determine that the Deployment Service has finished operating on existing artifacts in apps and domains?

**Pull request closed without merging**